### PR TITLE
Use SDL3 functions in more locations

### PR DESCRIPTION
**Pull request by madebr**
This slightly reduces the number of Windows-only apis.

Also untested.

I refactored the CMake script a bit because of [this Windows behavior](https://devblogs.microsoft.com/oldnewthing/20140321-00/?p=1433). Some symbols were not exported from the dll.

### CODE REVIEW

**Comment by foxtacles** [comment]
Looks good! I'll test it tomorrow but since all the SDL functions are API-compatible it should be fine.

**Comment by madebr** [comment]
> Looks good! I'll test it tomorrow but since all the SDL functions are API-compatible it should be fine.

I think the replacement of `timeGetTIme` with `SDL_Delay` is the biggest change. Both return times in ms.

File: LEGO1/lego/legoomni/src/build/legocarbuild.cpp [hunk 1]
```diff
@@ -559,7 +561,7 @@ MxResult LegoCarBuild::Tickle()
 	}
 
 	if (m_unk0x10a) {
-		DWORD time = timeGetTime();
+		Uint64 time = SDL_GetTicks();
 		DWORD dTime = (time - m_unk0x10c) / 100;
```
Comment (foxtacles) [inline LEGO1/lego/legoomni/src/build/legocarbuild.cpp:565]: I think it would be good if `DWORD` also becomes `Uint64` here.

**Review by foxtacles — APPROVED** [review]
Works as expected! Just added one comment. The fact that we now get the time since SDL initialized vs. time since Windows started is, as expected, not relevant.

File: LEGO1/lego/legoomni/src/build/legocarbuild.cpp (hunk 1 above)
Comment (madebr) [reply LEGO1/lego/legoomni/src/build/legocarbuild.cpp:565]: I hope the most significant bits are never used. If they are, then we have a big problem :)

```suggestion
		Uint64 dTime = (time - m_unk0x10c) / 100;
```

**Pull request merged**